gordo_components/server/server.py

```diff
@@ -117,6 +120,36 @@ def build_app():
     def _start_timer():
         g.start_time = timeit.default_timer()
 
+    @app.before_request
+    def _set_revision_and_collection_dir():
+        g.collection_dir = os.environ[
+            current_app.config["MODEL_COLLECTION_DIR_ENV_VAR"]
+        ]
+        g.current_revision = os.path.basename(g.collection_dir)
+
+        # If a specific revision was requested, update collection_dir
+        g.revision = request.args.get("revision") or request.headers.get("revision")
+        if g.revision:
+            g.collection_dir = os.path.join(g.collection_dir, "..", g.revision)
+            try:
+                os.listdir(g.collection_dir)  # List dir to ensure it exists
+            except FileNotFoundError:
```

Comment:
Could have done that, but I think we're leaning slightly towards the python's "asking forgiveness rather than permission" style. But I have no strong feelings towards this, aside from that it makes it slightly more explicit in the tests to verify what has happened in certain circumstances.